File: src/bdt/mars/project/v1/Producer.scala
package bdt.mars.project
import java.util.Properties
import org.apache.kafka.clients.producer._

object Producer extends App {
  val props = new Properties()
  props.put("bootstrap.servers", "kafka-66012d1-mum-6eb6.aivencloud.com:23237")

  props.put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
  props.put("value.serializer", "org.apache.kafka.common.serialization.StringSerializer")

  val producer = new KafkaProducer[String, String](props)

  val TOPIC = "election"

  for (i <- 1 to 50) {
    val record = new ProducerRecord(TOPIC, "key", s"hello $i")
    producer.send(record)
  }

  val record = new ProducerRecord(TOPIC, "key", "the end " + new java.util.Date)
  producer.send(record)

  producer.close()
  print("producer done")

}

File: src/bdt/mars/project/v1/Producer.java
package bdt.mars.project.v1;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerConfig;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.sql.Timestamp;
import java.util.Properties;

public class Producer {
	public static void main(String[] args) throws InterruptedException {
		Properties props = new Properties();
		props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
		props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
				"org.apache.kafka.common.serialization.StringSerializer");
		props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
				"org.apache.kafka.common.serialization.StringSerializer");
		KafkaProducer<String, String> producer = new KafkaProducer<String, String>(
				props);
		String[] candidates = { "Arya", "Jon", "Sansa", "Sersei", "Daenerys" };
		int csize = candidates.length;
		int usize = Names.giveMeSize();
		for (int i = 0; i < 100; i++) {
			ProducerRecord<String, String> data;
			double random_candidate = Math.random() * csize;
			double random_user = Math.random() * usize;
			Timestamp timestamp = new Timestamp(System.currentTimeMillis());
			data = new ProducerRecord<String, String>("election", 0,
					candidates[(int) random_candidate], new String(
							Names.giveMeName(random_user) + ","
									+ timestamp.getTime()));
			producer.send(data);
			Thread.sleep(1L);
		}
		producer.close();
	}
}